segment_coverage/segment_coverage/main.py

```diff
@@ -196,7 +207,12 @@ def run(self):
 					# based on common.segments.best_segments_by_start
 					# but more complicated to capture more detailed metrics
 					hour_path = os.path.join(self.base_dir, self.channel, quality, hour)
-					segment_names = [name for name in os.listdir(hour_path) if not name.startswith('.')]
+					try:
+						segment_names = [name for name in os.listdir(hour_path) if not name.startswith('.')]
+					except OSError as e:
+						if e.errno == errno.ENOENT:
+							self.logger.warning('{} does not exist'.format(hour_path))
```

Comment:
slightly more descriptive error would be `Hour {} was deleted between finding it and processing it, ignoring`